### PR TITLE
Log unpack error instead of failing silently

### PR DESCRIPTION
**Pull request by mohitkhullar**
Signed-off-by: Mohit Khullar <mohit.khullar@gmail.com>

### CODE REVIEW

File: plugins/newsql/newsql.c [hunk 1]
```diff
@@ -2139,8 +2139,14 @@ static CDB2QUERY *read_newsql_query(struct dbenv *dbenv,
         Pthread_mutex_unlock(&clnt->wait_mutex);
     }
 
-    if (!query) return NULL;
+    if (!query) {
+        logmsg(LOGMSG_ERROR, "%s:%d out of memory \n", __func__,__LINE__);
```
[the remark this answers is not in the document]
Comment (mohitkhullar) [reply plugins/newsql/newsql.c]: merged it with change below

File: plugins/newsql/newsql.c [hunk 2]
```diff
@@ -2139,9 +2139,12 @@ static CDB2QUERY *read_newsql_query(struct dbenv *dbenv,
         Pthread_mutex_unlock(&clnt->wait_mutex);
     }
 
-    if (!query) return NULL;
-    if (errno != 0) {
-        cdb2__query__free_unpacked(query, &pb_alloc);
+    if (!query || (errno != 0)) {
+        logmsg(LOGMSG_ERROR,
+               "%s:%d Error unpacking query error:%s\n",
```
Comment (riverszhang89) [inline plugins/newsql/newsql.c:2144]: I might be a bit nitpicking here: could we add a space between the colon and `%s`?

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/formatting-a-pr.html'>Use clang-format</a> or <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/19bb536/format.diff'>view diff</a> &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: Success &#10003;.</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/19bb536/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/19bb536/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2477'>Resubmit</a></p>

**Review by riverszhang89 — APPROVED** [review]
Good to merge!

**Review by roborivers — APPROVED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: Success &#10003;.</p><p>The first 10 failing tests are: <br>sp_queueodh_generated [core dumped]</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/f672e93/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/f672e93/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2477'>Resubmit</a></p>

**Review by roborivers — APPROVED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: Success &#10003;.</p><p>The first 10 failing tests are: <br>socksql_master_swings [core dumped]</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/f672e93/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/f672e93/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2477'>Resubmit</a></p>

**Review by roborivers — APPROVED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: Success &#10003;.</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/f672e93/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/f672e93/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2477'>Resubmit</a></p>

**Comment by mohitkhullar** [comment]
Thanks Rivers

**Pull request merged**